### PR TITLE
feat: rough in a user questionnaire form

### PR DESCRIPTION
**Pull request by anandaroop**
The type of this PR is: **Feat**

### Description

Adds a starter form for the user input  step of the discovery experiment.

All copy and suggestions are deliberately placeholder-y, based on the Notion [doc](https://www.notion.so/artsy/Curated-Discovery-5e43238042014fae853222582a35acd9) and follow-up conversations.

This is mostly just to give us a starting point to react to. 

https://github.com/artsy/force/assets/140521/70f4183a-6d3c-4939-b2f7-e512e6c5930e

### CODE REVIEW

File: src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import { Goals } from "./Components/Goals"
+
+export type State = {
+  goals: string[]
+  goalsFreeText: string
+  interests: string[]
+  interestsFreeText: string
+  artists: string[]
+  artistsFreeText: string
+}
```
Comment (anandaroop) [inline src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx:14]: Keeping the modeling simple, just a bunch of `string[]` arrays corresponding to the suggestion pills, and some `string`s corresponding to the freetext fields.

Upon form submission, we'd shape up this payload into something suitable for an LLM call.

(Posted a follow-up question [here](https://artsy.slack.com/archives/C06SSV1K10D/p1718722761894599))

File: src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    case "TOGGLE_GOAL":
+      const goals = state.goals.includes(action.goal)
+        ? state.goals.filter(goal => goal !== action.goal)
+        : [...state.goals, action.goal]
+      return { ...state, goals }
+    case "UNSURE_GOALS":
+      return { ...state, goals: [] }
+    case "SET_FREETEXT_GOALS":
+      return { ...state, goalsFreeText: action.text }
```
Comment (anandaroop) [inline src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx:53]: A lot of repetition across the three questions here, but the same idea for each: using state reducers to toggle or clear the suggestion pills; or to set the freetext field.

File: src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    </Box>
+  )
+}
+
+const PreviewState: FC<{
+  state: State
+}> = props => {
+  const { state } = props
+  return (
+    <pre
```
Comment (anandaroop) [inline src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx:116]: Just a temporary widget to display the payload we are building up. Will remove once we wire up the LLM step.

```json
{
  "goals": [
    "Expand my collection",
    "Buy some art"
  ],
  "goalsFreeText": "",
  "interests": [
    "Figurative Art"
  ],
  "interestsFreeText": "Very expensive old master paintings",
  "artists": [],
  "artistsFreeText": ""
}
```

**Comment by anandaroop** [comment]
Added https://github.com/artsy/force/pull/14036/commits/7984a312c06d8fe464a0acb0971fe687d67c2483 in response to [this comment](https://artsy.slack.com/archives/C06SSV1K10D/p1718722931484539?thread_ts=1718722761.894599&cid=C06SSV1K10D). 

(This marks some of the comments above as outdated, but the gist is still true)

File: src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx (hunk 3 above)
Comment (mc-jones) [reply src/Apps/ArtAdvisor/07-Curated-Discovery/App.tsx:96]: Very nice. Will for sure steal this in the future.

**Review by mc-jones — APPROVED** [review]
👏

**Pull request merged**